quicfire_tools/inputs.py

```diff
@@ -2205,81 +2376,449 @@ class QU_metparams(InputFile):
+        sensor_height : PositiveFloat
+            Optional. Updated height of the wind sensor in meters
+        x_location : PositiveInt
+            Optional. Updated location of the wind sensor in the x-direction (m)
+        y_location : PositiveInt
+            Optional. Updated location of the wind sensor in the y-direction (m)
+        """
+        sensor_number = len(self.sensor_array) + 1
+        sensor_name = "".join(["sensor", str(sensor_number)])
+        sensor = WindSensor(
+            name=sensor_name,
+            time_now=self.time_now,
+            x_location=x_location,
+            y_location=y_location,
+            wind_times=wind_times,
+            sensor_height=sensor_height,
+            wind_speeds=wind_speeds,
+            wind_directions=wind_directions,
         )
+        self.sensor_array.append(sensor)
+        self._update_wind_times()
+
+        return sensor
+
+    def update_sensor(
+        self,
+        sensor_name: str,
+        x_location: Optional[PositiveInt] = None,
+        y_location: Optional[PositiveInt] = None,
+        wind_speeds: Optional[Union[PositiveFloat, list(PositiveFloat)]] = None,
+        wind_directions: Optional[Union[NonNegativeInt, list(NonNegativeInt)]] = None,
+        wind_times: Optional[Union[NonNegativeInt, list(NonNegativeInt)]] = None,
+        sensor_height: Optional[PositiveFloat] = None,
+    ):
+        """
+        Update parameters for a single wind sensor. Any parameters not provided will
+        remain unchanged.
+
+        Parameters
+        ----------
+        update: str
+            Name of the wind sensor, e.g. "sensor1". Sensor must already exist in windsensors.
+        x_location : PositiveInt
+            Optional. Updated location of the wind sensor in the x-direction (m)
+        y_location : PositiveInt
+            Optional. Updated location of the wind sensor in the y-direction (m)
+        wind_speeds : PositiveFloat | list(PositiveFloat)
+            Optional. Updated wind speed or list of wind speeds in m/s
+        wind_directions: PositiveInt | list(PositiveInt)
+            Optional. Updated wind direction or list of wind directions in degrees. Use 0 for north.
+        wind_times : NonNegativeInt | list(NonNegativeInt)
+            Optional. Updated list of times for each windshift. Use 0 for single windshift. First value must be 0.
+        sensor_height : PositiveFloat
+            Optional. Updated height of the wind sensor in meters
+        """
+        for arg in [wind_times, wind_speeds, wind_directions]:
+            if isinstance(arg, (float, int)):
+                arg = [arg]
+        sensor_found = False
+        for sensor in self.sensor_array:
+            if sensor_name == sensor.name:
+                updater = sensor
+                sensor_found = True
+        if sensor_found == False:
+            raise AttributeError(f"{sensor_name} not found")
```

Comment:
I think we should split this logic up into a method. So it would result in something like:

    def _find_sensor_by_name(self, sensor_name: str) -> WindSensor | None:
        """
        Find and return sensor by name. Returns None if there is no matching sensor name.
        """
        for sensor in self.sensor_array:
            if sensor.name == sensor_name:
                return sensor
        return None

Then in update_sensor we'd have:

```python
updater = _find_sensor_by_name(sensor_name)
if not updater:
    raise AttributeError(f"{sensor_name} not found")